flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamOperatorStateHandler.java

```diff
@@ -282,8 +283,8 @@ public void setCurrentKey(Object key) {
 		if (keyedStateBackend != null) {
 			try {
 				// need to work around type restrictions
-				@SuppressWarnings("unchecked,rawtypes")
-				AbstractKeyedStateBackend rawBackend = (AbstractKeyedStateBackend) keyedStateBackend;
+				@SuppressWarnings("rawtypes")
+				CheckpointableKeyedStateBackend rawBackend = keyedStateBackend;
 
 				rawBackend.setCurrentKey(key);
```

Comment:
```suggestion
				((CheckpointableKeyedStateBackend) keyedStateBackend).setCurrentKey(key);
```
We could also remove "rawtypes" from the warning suppression of the method.

Reply:
In this line, we actually can not remove the "rawtypes" suppression. We can remove it from the method annotation though. I will do that.

Reply:
The suggested change looks ok in my local IntelliJ, could you double check? Thanks. :-)

Reply:
mine, shows it as a warning.

We are using the `CheckpointableKeyedStateBackend` as a raw type, without the key, so there should be a warning.

Reply:
True, but that's why we need to suppress the "unchecked" warning, rather than "rawtypes"? (In my local env it will also show the warning if I remove the "unchecked" suppression from the method signature)

Reply:
The "unchecked" suppressed the cast. The "rawtypes" suppresses the call to `setCurrentKey` on raw type.

Reply:
![suppress](https://user-images.githubusercontent.com/6242259/93879881-d4b72b00-fcdc-11ea-94c0-2030895acf4c.png)